test/Path/Gen.hs

```diff
@@ -76,3 +77,20 @@ genFilePath = listOf genPathyChar
 
 genPathyChar :: Gen Char
 genPathyChar = frequency [(2, arbitrary), (1, elements "./\\")]
+
+shrinkValidAbsFile :: Path Abs File -> [Path Abs File]
+shrinkValidAbsFile = shrinkValidWith parseAbsFile
+
+shrinkValidAbsDir :: Path Abs Dir -> [Path Abs Dir]
+shrinkValidAbsDir = shrinkValidWith parseAbsDir
+
+shrinkValidRelFile :: Path Rel File -> [Path Rel File]
+shrinkValidRelFile = shrinkValidWith parseRelFile
+
+shrinkValidRelDir :: Path Rel Dir -> [Path Rel Dir]
+shrinkValidRelDir = shrinkValidWith parseRelDir
+
+shrinkValidWith :: (FilePath -> Maybe (Path a b)) -> Path a b -> [Path a b]
+shrinkValidWith fun (Path s) = catMaybes $ do
+    candidate <- shrink s
+    pure $ fun candidate
```

Comment:
Good point.
I have to use an editor other than my standard editor because I use 'hindent' on Haskell code by default, so I added the newlines manually in what follows.